src/ui/handler/scroll_zoom.js

```diff
@@ -356,6 +375,46 @@ class ScrollZoomHandler {
     reset() {
         this._active = false;
     }
+
+    /**
+     * Returns the scroll zoom blocker alert's HTML element.
+     *
+     * @example
+     * // Change the scroll zoom blocker element's font size
+     * map.scrollZoom.getScrollZoomBlockerElement();
+     * scrollZoomBlockerElem.style.fontSize = "25px";
+     * @returns {HTMLElement} Returns container element.
+     */
+
+    getScrollZoomBlockerElement() {
+        if (this._container) return this._container;
+    }
+
+    _addScrollZoomBlocker() {
+        if (!this._map) return;
+        if (!this._container) {
+            this._container = DOM.create('div', 'mapboxgl-scroll-zoom-blocker-control', this._map._container);
+            const alertMessage = this._getBlockerAlertMessage();
+            const textNode = window.document.createTextNode(alertMessage);
+            this._container.appendChild(textNode);
+        }
+    }
+
+    _getBlockerAlertMessage() {
+        return this._map._getUIString('ScrollZoomBlocker.Message');
+    }
+
+    _showBlockerAlert() {
+        if (this._container.style.visibility === 'hidden') this._container.style.visibility = 'visible';
+        this._container.classList.remove('mapboxgl-scroll-zoom-blocker-control-fade');
+        this._container.style.opacity = '1';
+
+        this._container.ontransitionend = () => {
```

Comment:
It appears to me this will get executed as many times as `_showBlockerAlert` has been called. Do we need something like `once()`, i.e.

```js
const onTransitionEnd  = () => {
  ...
  _element.removeEventListener('transitionEnd', onTransitionEnd);
}
this._container.addEventListener('transitionend', onTransitionEnd);

Reply:
Ohhh wait, sorry, `ontransitionend = ` is the overwriting version, right, so that my comment is not true?

Reply:
Hi Ricky, can you clarify what you mean by the overwriting version? The ontransitionend is called every time showBlockerAlert is called. I am using ontransitionend to listen for when the opacity has finished transitioning to 1 before initiating the fade out.

Reply:
This was just my mixup. I forgot for a moment that `element.ontransitionend = ___` overwrites the event handler while `element.addEventListener('transitionend', ___)` would attach more and more to the same element each time this method is called, executing the callback as many times as you've attached it. The overwriting version is fine here. I believe the `transitionend` event handler _could_ be moved to an initialization method since we don't need to attach and reattach it every time we trigger the UI display. Since it was just a mixup for me to think anything was actually problematic, I don't think a change is necessary.